Examples/Tests/reduced_diags/inputs

```diff
@@ -89,8 +89,8 @@ FR_Min.reduced_function(x,y,z,Ex,Ey,Ez,Bx,By,Bz) = x*Ey*Bz
 FR_Min.reduction_type = Minimum
 FR_Integral.type = FieldReduction
 FR_Integral.intervals = 200
-FR_Integral.reduced_function(x,y,z,Ex,Ey,Ez,Bx,By,Bz) = "((y > 0)*(z < 0)) *
-                              (0.5*((Ex**2 + Ey**2 + Ez**2)*epsilon0+(Bx**2 + By**2 + Bz**2)/mu0))"
+FR_Integral.reduced_function(x,y,z,Ex,Ey,Ez,Bx,By,Bz) =
+    "if(y > 0, if(z < 0, 0.5*((Ex**2 + Ey**2 + Ez**2)*epsilon0+(Bx**2 + By**2 + Bz**2)/mu0), 0), 0)"
```

Comment:
@ax3l (Cc @WeiqunZhang)

Here I used two nested `if` because this
```
"if(y > 0 and z < 0, expr1, expr2)"
```
did not seem to work.

Reply:
It should work.  Let me check if there is a bug in the parser.

Reply:
Can you show me the exact line that does not work.  Just want to make sure there is no typo.

Reply:
I tried this:
```
FR_Integral.reduced_function(x,y,z,Ex,Ey,Ez,Bx,By,Bz) =
    "if(y > 0 and z < 0, 0.5*((Ex**2 + Ey**2 + Ez**2)*epsilon0+(Bx**2 + By**2 + Bz**2)/mu0), 0)"
```

Reply:
Thanks! Did you get a runtime error or a wrong results?

Reply:
I ran the test `reduced_diags` as `./run_test.sh reduced_diags`, with the modification above in its input file, and the test crashed.

In order to share some debugging information with you, I ran it also in DEBUG mode (by adding `DEBUG=TRUE` to the variable `add_to_c_make_command` in Regression/WarpX-tests.ini) and I got the following error file and backtrace: [reduced_diags.err.out.txt](https://github.com/ECP-WarpX/WarpX/files/6542754/reduced_diags.err.out.txt), [Backtrace.0.0.txt](https://github.com/ECP-WarpX/WarpX/files/6542756/Backtrace.0.0.txt).

Reply:
The issue is the Parser is given a wrong string because all white spaces are removed by https://github.com/ECP-WarpX/WarpX/blob/c18569f0a67bdf5f33c8b0ecab147dfb7a52bff5/Source/Diagnostics/ReducedDiags/FieldReduction.cpp#L46

So `if(y > 0 and z < 0, exp, 0)` becomes `if(y>0andz<0,exp,0)`.  We should remove the code that removes the spaces.

Reply:
We could also remove the line that removes `\n` because `makeParser` will do it too.

Reply:
Thank you, @WeiqunZhang! I can try to remove that line directly here and see if that works.